app/controllers/view_components_controller.rb

```diff
@@ -59,4 +61,8 @@ def set_locale
   def prepend_application_view_paths
     prepend_view_path Rails.root.join("app/views") if defined?(Rails.root)
   end
+
+  def url_helpers
+    Rails.application.routes.url_helpers
+  end
```

Comment:
Don't all controllers have url_helpers already? I'm surprised we need this.

Reply:
Surprisingly enough, Rails::ApplicationController (parent class) does not have the url helpers. 

Maybe this is a sign that we should start to decouple from rails internals, like the Rails::ApplicationController and start rolling our own standalone controller (?)